File: financeApp-backend-feature-admin-group-management/app/Models/Incoming.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class Incoming extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'user_id',
        'organization_id',
        'department_id',
        'description',
        'amount_usd',
        'incoming_date',
        'sync_status',
        'synced_at',
    ];

    protected $casts = [
        'incoming_date' => 'date',
        'synced_at' => 'datetime',
        'amount_usd' => 'decimal:2',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function organization(): BelongsTo
    {
        return $this->belongsTo(Organization::class);
    }

    public function department(): BelongsTo
    {
        return $this->belongsTo(Department::class);
    }
}

File: financeApp-backend-feature-admin-group-management/financeApp-supabase-version/lib/features/incoming/domain/entities/incoming.dart
import 'package:equatable/equatable.dart';

class Incoming extends Equatable {
  final int? id;
  final int userId;
  final String description;
  final double amountUsd;
  final DateTime transactionDate;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const Incoming({
    this.id,
    required this.userId,
    required this.description,
    required this.amountUsd,
    required this.transactionDate,
    required this.createdAt,
    this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        description,
        amountUsd,
        transactionDate,
        createdAt,
        updatedAt,
      ];
}
